### PR TITLE
feat(libs): add `res-to-ts` library

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview

Allows packaging resource files in a way that TypeScript will include with the build. This will be useful for bundling scanning template images for the browser and fixtures for tests.

## Demo Video or Screenshot
Example output for `ballot-interpreter-nh/src/data/templates/oval.png`:
```ts
/* Generated by res-to-ts. DO NOT EDIT */
/* eslint-disable */
/* istanbul ignore file */
import { createCanvas, Image, ImageData, loadImage } from 'canvas';

/**
 * Data of ../ballot-interpreter-nh/src/data/templates/oval.png encoded as base64.
 */
const resourceDataBase64 = 'iVBORw0KGgoAAAANSUhEUgAAAA8AAAAKCAIAAADkeZOuAAAAAXNSR0IArs4c6QAAAMJlWElmTU0AKgAAAAgABwESAAMAAAABAAEAAAEaAAUAAAABAAAAYgEbAAUAAAABAAAAagEoAAMAAAABAAIAAAExAAIAAAARAAAAcgEyAAIAAAAUAAAAhIdpAAQAAAABAAAAmAAAAAAAAABIAAAAAQAAAEgAAAABUGl4ZWxtYXRvciAzLjkuOQAAMjAyMjowMjoxNCAxNjowMjoyNQAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAD6ADAAQAAAABAAAACgAAAAAlYIviAAAACXBIWXMAAAsTAAALEwEAmpwYAAADqGlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iWE1QIENvcmUgNi4wLjAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyIKICAgICAgICAgICAgeG1sbnM6ZXhpZj0iaHR0cDovL25zLmFkb2JlLmNvbS9leGlmLzEuMC8iCiAgICAgICAgICAgIHhtbG5zOnhtcD0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wLyI+CiAgICAgICAgIDx0aWZmOkNvbXByZXNzaW9uPjA8L3RpZmY6Q29tcHJlc3Npb24+CiAgICAgICAgIDx0aWZmOlJlc29sdXRpb25Vbml0PjI8L3RpZmY6UmVzb2x1dGlvblVuaXQ+CiAgICAgICAgIDx0aWZmOlhSZXNvbHV0aW9uPjcyPC90aWZmOlhSZXNvbHV0aW9uPgogICAgICAgICA8dGlmZjpZUmVzb2x1dGlvbj43MjwvdGlmZjpZUmVzb2x1dGlvbj4KICAgICAgICAgPHRpZmY6T3JpZW50YXRpb24+MTwvdGlmZjpPcmllbnRhdGlvbj4KICAgICAgICAgPGV4aWY6UGl4ZWxYRGltZW5zaW9uPjE1PC9leGlmOlBpeGVsWERpbWVuc2lvbj4KICAgICAgICAgPGV4aWY6Q29sb3JTcGFjZT4xPC9leGlmOkNvbG9yU3BhY2U+CiAgICAgICAgIDxleGlmOlBpeGVsWURpbWVuc2lvbj4xMDwvZXhpZjpQaXhlbFlEaW1lbnNpb24+CiAgICAgICAgIDx4bXA6Q3JlYXRvclRvb2w+UGl4ZWxtYXRvciAzLjkuOTwveG1wOkNyZWF0b3JUb29sPgogICAgICAgICA8eG1wOk1vZGlmeURhdGU+MjAyMi0wMi0xNFQxNjowMjoyNTwveG1wOk1vZGlmeURhdGU+CiAgICAgIDwvcmRmOkRlc2NyaXB0aW9uPgogICA8L3JkZjpSREY+CjwveDp4bXBtZXRhPgrtRHlzAAABFklEQVQYGWWRycmEQBCF1XHBEBQEA9CTGJQJeBIMQTybgGEoRuECXlyYYUTQg4jbPP8CL3+Dj9dd3V+9Qva6LuZvfT6fPM+7ruN5nmXZbdter5ckSaZpqqoqiuJ5ngxu7/seRVGWZfM8Y4vTR/GmLEvHcb7f783FFwRBXdcwWMdxkAEChhTG930osyxLHMdwhITBApIM6bquaJumKd/3vW3bFB01JIZHdCj6IDoOERqmbVsOGFCh9OC/3sMxDMdxt2qaVhSFIAgPGKegQsEjBev9fuu6zsmyjExN01CG5x4ZKNhgua5rWRZDA3melyTJNE3oQOOSolpVVRiG4zjezamMjsMwoICh4ZESVCQB1TAMRVHo7/wAxBgJn1bwpVUAAAAASUVORK5CYII=';

/**
 * MIME type of ../ballot-interpreter-nh/src/data/templates/oval.png.
 */
export const mimeType = 'image/png';

/**
 * Convert to a `data:` URL of ../ballot-interpreter-nh/src/data/templates/oval.png, suitable for embedding in HTML.
 */
export function asDataUrl() {
  return `data:${mimeType};base64,${resourceDataBase64}`;
}

/**
 * Raw data of ../ballot-interpreter-nh/src/data/templates/oval.png.
 */
export function asBuffer(): Buffer {
  return Buffer.from(resourceDataBase64, 'base64');
}

/**
 * Converts ../ballot-interpreter-nh/src/data/templates/oval.png to an `Image`.
 */
export async function asImage(): Promise<Image> {
  return await loadImage(asDataUrl());
}

/**
 * Converts ../ballot-interpreter-nh/src/data/templates/oval.png to an `ImageData`.
 */
export async function asImageData(): Promise<ImageData> {
  const image = await asImage();
  const canvas = createCanvas(image.width, image.height);
  const context = canvas.getContext('2d');
  context.drawImage(image, 0, 0);
  return context.getImageData(0, 0, image.width, image.height);
}
```

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [x] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by carolinemodic** [comment]
Would this be usable if we wanted to include images (like scanning ballot images) in libs/fixtures?

**Review by carolinemodic — COMMENTED** [review]
Cool! I'm assuming the second commit is included by mistake it seems unrelated? 

We should also migrate libs/fixtures to use this for the sems/cvr files. (not in this PR just in general)

File: libs/res-to-ts/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# res-to-ts
+
+Converts a resource file to a TypeScript file. If you need to include a non-code resource such as an image, CSV, or XML file with your package then you can use `res-to-ts` to do so.
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply libs/res-to-ts/README.md:3]: It actually does have support for `jsonl` already 😉

**Comment by eventualbuddha** [comment]
> Cool! I'm assuming the second commit is included by mistake it seems unrelated?
Fixed.

> We should also migrate libs/fixtures to use this for the sems/cvr files. (not in this PR just in general)
Yep, I started doing this as a way to vet this library. The `jsonl` support comes from making sure it works with `libs/fixtures`. I'll throw another PR for that up soon.

File: libs/res-to-ts/src/mime.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { getType } from 'mime';
+import { extname } from 'path';
+
+/**
+ * Gets the MIME type of files not supported by `mime`.
+ */
+function getFallbackMimeType(path: string): string {
+  const ext = extname(path);
+  if (ext === '.jsonl') {
+    return 'application/jsonlines';
```
Comment (eventualbuddha) [inline libs/res-to-ts/src/mime.ts:10]: @carolinemodic `jsonl` support

File: libs/res-to-ts/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (carolinemodic) [reply libs/res-to-ts/README.md:3]: oh haha nice

**Review by carolinemodic — APPROVED** [review]

**Pull request merged**